server/game/core/ability/TriggeredAbility.ts

```diff
@@ -112,6 +115,18 @@ export default class TriggeredAbility extends CardAbility {
         });
     }
 
+    public override checkGameActionsForPotential(context) {
+        const mustChangeGameState = !!this.properties.ifYouDo || !!this.properties.ifYouDoNot;
+
+        return this.immediateEffect.hasLegalTarget(context, {}, mustChangeGameState);
+    }
+
+    public override buildTargetResolver(name: any, properties: any) {
+        const propsMustChangeGameState = Object.assign({ mustChangeGameState: this.mustChangeGameState }, properties);
```

Comment:
Thought: This is totally fine and no need to change it. Just want to call out you can use the less verbose spreading as an alternative just in case you aren't familiar with it.

```suggestion
        const propsMustChangeGameState = { mustChangeGameState: this.mustChangeGameState, ...properties};
```

Reply:
Oh yeah good call, I was using Object.assign exclusively for a while and only just recently learned about the spread operator. Definitely want to use it more often now